### PR TITLE
Add the data_manager_malt_index_builder tool

### PR DESCRIPTION
**Pull request by gregvonkuster**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by bgruening** [comment]
This is included in the malt-build shell script:

```
read_vmoptions() {
  vmoptions_file=`eval echo "$1" 2>/dev/null`
  if [ ! -r "$vmoptions_file" ]; then
    vmoptions_file="$prg_dir/$vmoptions_file"
  fi
  if [ -r "$vmoptions_file" ] && [ -f "$vmoptions_file" ]; then
    exec 8< "$vmoptions_file"
    while read cur_option<&8; do
      is_comment=`expr "W$cur_option" : 'W *#.*'`
      if [ "$is_comment" = "0" ]; then 
        vmo_classpath=`expr "W$cur_option" : 'W *-classpath \(.*\)'`
        vmo_classpath_a=`expr "W$cur_option" : 'W *-classpath/a \(.*\)'`
        vmo_classpath_p=`expr "W$cur_option" : 'W *-classpath/p \(.*\)'`
        vmo_include=`expr "W$cur_option" : 'W *-include-options \(.*\)'`
        if [ ! "W$vmo_include" = "W" ]; then
            if [ "W$vmo_include_1" = "W" ]; then
              vmo_include_1="$vmo_include"
            elif [ "W$vmo_include_2" = "W" ]; then
              vmo_include_2="$vmo_include"
            elif [ "W$vmo_include_3" = "W" ]; then
              vmo_include_3="$vmo_include"
            fi
        fi
        if [ ! "$vmo_classpath" = "" ]; then
          local_classpath="$i4j_classpath:$vmo_classpath"
        elif [ ! "$vmo_classpath_a" = "" ]; then
          local_classpath="${local_classpath}:${vmo_classpath_a}"
        elif [ ! "$vmo_classpath_p" = "" ]; then
          local_classpath="${vmo_classpath_p}:${local_classpath}"
        elif [ "W$vmo_include" = "W" ]; then
          needs_quotes=`expr "W$cur_option" : 'W.* .*'`
          if [ "$needs_quotes" = "0" ]; then 
            vmoptions_val="$vmoptions_val $cur_option"
          else
            if [ "W$vmov_1" = "W" ]; then
              vmov_1="$cur_option"
            elif [ "W$vmov_2" = "W" ]; then
              vmov_2="$cur_option"
            elif [ "W$vmov_3" = "W" ]; then
              vmov_3="$cur_option"
            elif [ "W$vmov_4" = "W" ]; then
              vmov_4="$cur_option"
            elif [ "W$vmov_5" = "W" ]; then
              vmov_5="$cur_option"
            fi
          fi
        fi
      fi
    done
    exec 8<&-
    if [ ! "W$vmo_include_1" = "W" ]; then
      vmo_include="$vmo_include_1"
      unset vmo_include_1
      read_vmoptions "$vmo_include"
    fi
    if [ ! "W$vmo_include_2" = "W" ]; then
      vmo_include="$vmo_include_2"
      unset vmo_include_2
      read_vmoptions "$vmo_include"
    fi
    if [ ! "W$vmo_include_3" = "W" ]; then
      vmo_include="$vmo_include_3"
      unset vmo_include_3
      read_vmoptions "$vmo_include"
    fi
  fi
}
```
So it also looks like we can pass such an option file as parameter? Or pass an empty file and let handle Java the memory setting with the global fallback `export _JAVA_OPTIONS="-Xmx40G -Xms1G"`?

**Comment by gregvonkuster** [comment]
> This is included in the malt-build shell script:
> 
> ```
> read_vmoptions() {
>   vmoptions_file=`eval echo "$1" 2>/dev/null`
>   if [ ! -r "$vmoptions_file" ]; then
>     vmoptions_file="$prg_dir/$vmoptions_file"
>   fi
>   if [ -r "$vmoptions_file" ] && [ -f "$vmoptions_file" ]; then
>     exec 8< "$vmoptions_file"
>     while read cur_option<&8; do
>       is_comment=`expr "W$cur_option" : 'W *#.*'`
>       if [ "$is_comment" = "0" ]; then 
>         vmo_classpath=`expr "W$cur_option" : 'W *-classpath \(.*\)'`
>         vmo_classpath_a=`expr "W$cur_option" : 'W *-classpath/a \(.*\)'`
>         vmo_classpath_p=`expr "W$cur_option" : 'W *-classpath/p \(.*\)'`
>         vmo_include=`expr "W$cur_option" : 'W *-include-options \(.*\)'`
>         if [ ! "W$vmo_include" = "W" ]; then
>             if [ "W$vmo_include_1" = "W" ]; then
>               vmo_include_1="$vmo_include"
>             elif [ "W$vmo_include_2" = "W" ]; then
>               vmo_include_2="$vmo_include"
>             elif [ "W$vmo_include_3" = "W" ]; then
>               vmo_include_3="$vmo_include"
>             fi
>         fi
>         if [ ! "$vmo_classpath" = "" ]; then
>           local_classpath="$i4j_classpath:$vmo_classpath"
>         elif [ ! "$vmo_classpath_a" = "" ]; then
>           local_classpath="${local_classpath}:${vmo_classpath_a}"
>         elif [ ! "$vmo_classpath_p" = "" ]; then
>           local_classpath="${vmo_classpath_p}:${local_classpath}"
>         elif [ "W$vmo_include" = "W" ]; then
>           needs_quotes=`expr "W$cur_option" : 'W.* .*'`
>           if [ "$needs_quotes" = "0" ]; then 
>             vmoptions_val="$vmoptions_val $cur_option"
>           else
>             if [ "W$vmov_1" = "W" ]; then
>               vmov_1="$cur_option"
>             elif [ "W$vmov_2" = "W" ]; then
>               vmov_2="$cur_option"
>             elif [ "W$vmov_3" = "W" ]; then
>               vmov_3="$cur_option"
>             elif [ "W$vmov_4" = "W" ]; then
>               vmov_4="$cur_option"
>             elif [ "W$vmov_5" = "W" ]; then
>               vmov_5="$cur_option"
>             fi
>           fi
>         fi
>       fi
>     done
>     exec 8<&-
>     if [ ! "W$vmo_include_1" = "W" ]; then
>       vmo_include="$vmo_include_1"
>       unset vmo_include_1
>       read_vmoptions "$vmo_include"
>     fi
>     if [ ! "W$vmo_include_2" = "W" ]; then
>       vmo_include="$vmo_include_2"
>       unset vmo_include_2
>       read_vmoptions "$vmo_include"
>     fi
>     if [ ! "W$vmo_include_3" = "W" ]; then
>       vmo_include="$vmo_include_3"
>       unset vmo_include_3
>       read_vmoptions "$vmo_include"
>     fi
>   fi
> }
> ```
> 
> So it also looks like we can pass such an option file as parameter? Or pass an empty file and let handle Java the memory setting with the global fallback `export _JAVA_OPTIONS="-Xmx40G -Xms1G"`?

Passing the .vmoptions file as a command line parameter is not possible.  I'm testing to see if something like this might work ````export _JAVA_OPTIONS="-Xmx40G -Xms1G"````

**Comment by gregvonkuster** [comment]
This setting ```export _JAVA_OPTIONS="-Xmx40G -Xms1G"``` does seem to override the content of the .vmoptions files, so the job_conf setting should work.

File: data_managers/data_manager_malt_index_builder/data_manager/malt_index_builder.xml [hunk 1]
```diff
@@ -0,0 +1,128 @@
+<tool id="data_manager_malt_index_builder" name="MALT index builder" tool_type="manage_data" version="0.5.3+galaxy0" profile="21.01">
```
[the remark this answers is not in the document]
Comment (gregvonkuster) [reply data_managers/data_manager_malt_index_builder/data_manager/malt_index_builder.xml:1]: I don't think macros are typically used for data managers.  It looks like the only exception is the rna_star_index_builder which links to a shared macro:
```
macros.xml -> ../../../tools/rgrnastar/macros.xml
```

[the remark this answers is not in the document]
Comment (gregvonkuster) [reply data_managers/data_manager_malt_index_builder/data_manager/malt_index_builder.xml:1]: Regarding the different versions, for some reason the bioconda recipe (previously contributed by someone else) is using the version string without the patch release decimal due to a sorting issue.  I have no idea why this is necessary.  Here's a snippet of the meta.yaml file:
```
package:
    name: malt
    version: 0.53 # Keep without patch release decimal. Versioning style changed in 0.5 causing latest-version sorting problems when installing from bioconda.  
build:
    noarch: generic
    number: 0
source:
    md5: bf8a3ecf39d66426a3dead1d2addd1ae
    url: https://software-ab.informatik.uni-tuebingen.de/download/malt/MALT_unix_0_5_3.sh
```

File: data_managers/data_manager_malt_index_builder/tool-data/malt_indices.loc.sample [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# then the malt_index.loc entry would look like this:
+#
+# phiX174   phiX   phiX Pretty   /depot/data2/galaxy/phiX/base/phiX.fa
+#
+# and your /depot/data2/galaxy/phiX/base/ directory
+# would contain phiX.fa.* files:
+#
+#-rw-r--r--  1 galaxy    universe 830134 2005-09-13 10:12 phiX.fa
+#...etc...
+
```
[the remark this answers is not in the document]
Comment (gregvonkuster) [reply data_managers/data_manager_malt_index_builder/tool-data/malt_indices.loc.sample]: I've corrected the file contents and added an example row.

**Comment by gregvonkuster** [comment]
I believe this is ready for final review / merge.   Thanks!

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
@gregvonkuster can you let me know which indices and which parameters you will use/create?

**Comment by gregvonkuster** [comment]
> @gregvonkuster can you let me know which indices and which parameters you will use/create?

Absolutely! Thanks @bgruening

**Pull request merged**